src/hardware.c

```diff
@@ -962,23 +953,19 @@ void Video_Detect() {
                 val |= I2C_Read16(ADDR_TC3587, 0x006A);
                 val |= I2C_Read16(ADDR_TC3587, 0x006E);
             }
-            if (val) {
-                LED_BLUE_ON;
-                led_status = ON;
-            } else {
-                LED_BLUE_OFF;
-                led_status = OFF;
-            }
+            if (val)
+                cameraLost = 0;
```

Comment:
Maybe `cameraLost = !val;` is enough... ?

Reply:
No, `val` is not only 0 and 1.

Reply:
does not matter, same result but no branches. `!` just inverse the val as you like to do here :)
I haven't check the assembly output so luckily the compiler could optimize this...